### PR TITLE
Unify molecule splitting and alignment

### PR DESCRIPTION
**Pull request by apayne97**
## Description
Dependent on #272 to be merged first!

Add testing functionality for the various functions we've been using to split the molecules up as we process them, and replace them with a single function that does that.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [x] remove align_receptor and prep_receptor
  - [x] make function to get the chains of any component of interest
  - [x] simplify the splitting functions and put them in the modeling module

### CODE REVIEW

File: asapdiscovery-data/asapdiscovery/data/openeye.py [hunk 1]
```diff
@@ -416,186 +417,6 @@ def openeye_perceive_residues(prot: oechem.OEGraphMol) -> oechem.OEGraphMol:
     return prot
 
 
-def split_openeye_mol(complex_mol, lig_chain="A", prot_cutoff_len=10):
```
Comment (hmacdope) [inline asapdiscovery-data/asapdiscovery/data/openeye.py:419]: Yep i think this is a good idea to move this out of the core `openeye` functionality, just make sure to change the call sites as you have already started doing. :+1:

File: asapdiscovery-modeling/asapdiscovery/modeling/schema.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from enum import Enum
+
+from pydantic import BaseModel, Field
+
+
+class MoleculeComponents(str, Enum):
```
Comment (hmacdope) [inline asapdiscovery-modeling/asapdiscovery/modeling/schema.py:6]: Very nice forward thinking!

File: asapdiscovery-modeling/asapdiscovery/modeling/tests/test_modeling_utils.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    else:
+        assert find_protein_chains(split_mol) == []
+
+
+@pytest.mark.parametrize("ligand_chain", ["A", "B"])
+@pytest.mark.parametrize(
+    "components",
+    [["ligand"], ["protein", "ligand"], ["protein", "ligand", "water"]],
+)
+def test_pdb_ligand_splitting(sars_oe, local_path, files, ligand_chain, components):
```
Comment (hmacdope) [inline asapdiscovery-modeling/asapdiscovery/modeling/tests/test_modeling_utils.py:63]: This is really cool.

File: asapdiscovery-modeling/asapdiscovery/modeling/tests/test_modeling_utils.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            f"split_test_{'-'.join(components)}_lig{ligand_chain}.pdb",
+        ),
+    )
+
+
+@pytest.mark.parametrize(
+    "components",
+    [["protein"], ["protein", "ligand"], ["protein", "ligand", "water"]],
+)
+@pytest.mark.parametrize("protein_chains", [["A"], ["B"], ["A", "B"]])
```
Comment (hmacdope) [inline asapdiscovery-modeling/asapdiscovery/modeling/tests/test_modeling_utils.py:84]: Does this just test that you can then save the split?

File: asapdiscovery-modeling/asapdiscovery/modeling/tests/test_modeling_utils.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        protein_chains=protein_chains,
+    )
+    complex = split_openeye_mol(sars_oe, molfilter)
+    assert find_protein_chains(complex) == protein_chains
+
+    save_openeye_pdb(
+        complex,
+        Path(
+            local_path,
+            f"split_test_{'-'.join(components)}_prot{''.join(protein_chains)}.pdb",
```
Comment (hmacdope) [inline asapdiscovery-modeling/asapdiscovery/modeling/tests/test_modeling_utils.py:97]: Probably should clean up the resulting file.

Can either do that with a directory parameter that acts as a context or  just with a direct rm. 

directory context looks like:

```python
@pytest.fixture()
def make_output_dir_and_cleanup():
    # create output dir
    os.makedirs("./outputs", exist_ok=True)
    yield # enters into the test
    # clean up afterwards
    shutil.rmtree("./outputs")
    ```

**Review by hmacdope — CHANGES_REQUESTED** [review]
Apologies for the premature review, this looks great! Just though I would add in some thoughts.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Apologies for the premature review, this looks great! Just though I would add in some thoughts.

**Comment by hmacdope** [comment]
@apayne97 I am going to add to 0.1.0. :smiley:

File: asapdiscovery-data/asapdiscovery/data/openeye.py (hunk 1 above)
Comment (apayne97) [reply asapdiscovery-data/asapdiscovery/data/openeye.py:419]: I'm using PyCharm to "refactor" the function and move it, which automatically moves all the imports, and I'm also double checking with checking for usage. Hopefully I'll catch everything!

File: asapdiscovery-modeling/asapdiscovery/modeling/schema.py (hunk 2 above)
Comment (apayne97) [reply asapdiscovery-modeling/asapdiscovery/modeling/schema.py:6]: thanks!!

File: asapdiscovery-modeling/asapdiscovery/modeling/tests/test_modeling_utils.py (hunk 5 above)
Comment (apayne97) [reply asapdiscovery-modeling/asapdiscovery/modeling/tests/test_modeling_utils.py]: Yeah this was just to be able to check it manually....I've changed the behaviour to only save if a local path is passed, which I think is the behaviour I want

**Review by hmacdope — APPROVED** [review]
Thanks @apayne97

**Review by hmacdope — APPROVED** [review]
Thanks @apayne97

**Pull request merged**